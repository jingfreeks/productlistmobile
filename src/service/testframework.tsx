/* eslint-disable @typescript-eslint/no-explicit-any */
import React,{PropsWithChildren} from 'react';
import {render} from '@testing-library/react-native';
import type {RenderOptions} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import type {AppStore, RootState} from '@/config/store';
import {setupStore} from '@/config/store'
// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
