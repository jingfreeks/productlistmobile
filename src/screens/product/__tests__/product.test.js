import React from 'react';
import { renderWithProviders } from "../../../service/testframework";
import {Product} from '../index';


describe('Product Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Product />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
