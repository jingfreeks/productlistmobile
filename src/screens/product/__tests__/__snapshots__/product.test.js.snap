// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Screen Should work as expected to get snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginBottom": 10,
      },
    ]
  }
>
  <ActivityIndicator
    color="#0000ff"
    size="large"
  />
</RCTSafeAreaView>
`;
